dissect/target/plugins/os/unix/cronjobs.py

```diff
@@ -29,82 +31,116 @@
 )
 
+RE_CRONJOB = re.compile(
+    r"""
+        ^
+        (?P<minute>\S+)
+        \s+
+        (?P<hour>\S+)
+        \s+
+        (?P<day>\S+)
+        \s+
+        (?P<month>\S+)
+        \s+
+        (?P<weekday>\S+)
+        \s+
+        (?P<command>.+)
+        $
+    """,
+    re.VERBOSE,
+)
+RE_ENVVAR = re.compile(
+    r"""
+        ^
+        ([a-zA-Z_]+[a-zA-Z[0-9_])=(.*)
+    """,
+    re.VERBOSE,
+)
+
 
 class CronjobPlugin(Plugin):
     """Unix cronjob plugin."""
 
+    CRONTAB_DIRS = [
+        "/var/cron/tabs",
+        "/var/spool/cron",
+        "/var/spool/cron/crontabs",
+        "/etc/cron.d",
+        "/usr/local/etc/cron.d",  # FreeBSD
+    ]
+
+    CRONTAB_FILES = [
+        "/etc/crontab",
+        "/etc/anacrontab",
+    ]
+
+    def __init__(self, target):
+        super().__init__(target)
+        self.crontabs = list(self.find_crontabs())
+
     def check_compatible(self) -> None:
-        pass
+        if not self.crontabs:
+            raise UnsupportedPluginError("No crontab(s) found on target")
 
-    def parse_crontab(self, file_path) -> Iterator[CronjobRecord | EnvironmentVariableRecord]:
-        for line in file_path.open("rt"):
-            line = line.strip()
-            if line.startswith("#") or not len(line):
+    def find_crontabs(self) -> Iterator[Path]:
+        for crontab_dir in self.CRONTAB_DIRS:
+            if not (dir := self.target.fs.path(crontab_dir)).exists():
                 continue
 
-            match = re.search(r"^(\S+)\s+(\S+)\s+(\S+)\s+(\S+)\s+(\S+)\s+(\S+)\s+(.*)$", line)
-            if match:
-                usr = match.group(6)
-                cmd = match.group(7)
-                if not str(file_path).startswith("/etc/crontab") or not str(file_path).startswith("/etc/cron.d"):
-                    cmd = usr + " " + cmd
-                    usr = ""
-
-                yield CronjobRecord(
-                    minute=match.group(1),
-                    hour=match.group(2),
-                    day=match.group(3),
-                    month=match.group(4),
-                    weekday=match.group(5),
-                    user=usr,
-                    command=cmd,
-                    source=file_path,
-                    _target=self.target,
-                )
-
-            s = re.search(r"^([a-zA-Z_]+[a-zA-Z[0-9_])=(.*)", line)
-            if s:
-                yield EnvironmentVariableRecord(
-                    key=s.group(1),
-                    value=s.group(2),
-                    source=file_path,
-                    _target=self.target,
-                )
-
-    @export(record=[CronjobRecord, EnvironmentVariableRecord])
+            for file in dir.iterdir():
+                if file.resolve().is_file():
+                    yield file
+
+        for crontab_file in self.CRONTAB_FILES:
+            if (file := self.target.fs.path(crontab_file)).exists():
+                yield file
+
+    @export(record=get_args([CronjobRecord, EnvironmentVariableRecord]))
```

Comment:
```suggestion
    @export(record=[CronjobRecord, EnvironmentVariableRecord])
```
The `get_args` is only for when you re-use a `Union` type.